src/api/fleet_addon_config.rs

```diff
@@ -63,18 +63,96 @@ pub struct ClusterConfig {
     /// Setting to disable setting owner references on the created resources
     pub set_owner_references: Option<bool>,
 
+    /// Specifies a name suffix for the fleet cluster
+    pub naming: NamingStrategy,
+
     #[cfg(feature = "agent-initiated")]
     /// Prepare initial cluster for agent initiated connection
     pub agent_initiated: Option<bool>,
 }
 
+/// NamingStrategy is controlling Fleet cluster naming
+#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema, Default)]
+pub struct NamingStrategy {
+    /// Specify a prefix for the Cluster name, applied to created Fleet cluster
+    pub prefix: Option<String>,
+    /// Specify a suffix for the Cluster name, applied to created Fleet cluster
+    pub suffix: Option<String>,
+}
+
 impl Default for ClusterConfig {
     fn default() -> Self {
         Self {
             set_owner_references: Some(true),
+            naming: Default::default(),
             enabled: Some(true),
             #[cfg(feature = "agent-initiated")]
             agent_initiated: Some(true),
         }
     }
 }
+
+impl NamingStrategy {
+    pub fn apply(&self, name: Option<String>) -> Option<String> {
+        name.map(|name| match self.prefix.clone() {
+            Some(prefix) => prefix + &name,
+            None => name,
+        })
+        .map(|name| match self.suffix.clone() {
+            Some(suffix) => name + &suffix,
+            None => name,
+        })
+    }
+}
```

Comment:
How about avoiding cloning always (even if it is None) with matching on a reference to the prefix and suffix instead? 

```
impl NamingStrategy {
    pub fn apply(&self, name: Option<String>) -> Option<String> {
        name.map(|name| match self.prefix.as_ref() {
            Some(prefix) => prefix.clone() + &name,
            None => name,
        })
        .map(|name| match self.suffix.as_ref() {
            Some(suffix) => name + suffix,
            None => name,
        })
    }
}
```

Reply:
Generally I don’t think it matters, especially here. I’d update the code, but readability in this stage of the project matters more than performance. There will be always room for refactor, if this will become needed.